src/main/webapp/app/complaints/complaints-for-tutor/complaints-for-tutor.component.ts

```diff
@@ -17,7 +17,7 @@ import { Submission } from 'app/entities/submission.model';
     providers: [],
 })
 export class ComplaintsForTutorComponent implements OnInit {
-    @Input() complaint: Complaint;
+    @Input() complaint: Complaint | undefined;
```

Comment:
This component seems to always need a complaint. So instead of making the input possibly undefined, we could make sure that all code that uses the component actually passes a valid complaint object

Reply:
Or is there are case in which the complaint could actually be undefined?

Reply:
just logically looking at the parent components, it is defined as optional. so I would keep it optional here aswell

Reply:
I'm not sure what you mean. The whole component is only used once and already checks if the complaint is available

<img width="657" alt="image" src="https://user-images.githubusercontent.com/744067/115142115-43c0ec00-a040-11eb-9886-37c4a0ed1833.png">